Rahav/cert 1850 external documentation (#98)
Certora/Documentation: docs/prover/cli/options.md

If we have a Solidity file `Bank.sol`, with a contract named `Bank` inside it, and a specification file called `Bank.spec`, the run command would be:  
`certoraRun Bank.sol --verify Bank:Bank.spec`

> ### `--assert_contract`

Reviewer: isn't that assert_contracts? I'm always confusing this one

Author: There is assert_contracts but we marked is as "internal" so it should not be in this doc

Author: fixed (apparently  you are not the only one that get confused here)

**What does it do?**  
Gets a list of arguments to pass to the Solidity compiler. The arguments will be passed as is, without any formatting, in the same order.  

> arguments that are defined as standalone flags cannot be defined in `--solc_args`, including the solc flags  `--optimize`, 

Reviewer 2: `solc` fails spell checking; you should either use "the Solidity compiler" or
```
`solc`
```

Author: done

**When to use it?**  
When we want to activate in the solidity compiler the opcode-based optimizer for the generated bytecode and control the 
number of times the optimizer will be activated (if no value is set, the compiler's default is 200 runs)

**Example**
`certoraRun Bank.sol --verify Bank:Bank.spec --solc_optimize 300`

### `--solc_via_ir`

**What does it do?**  
Passes the value of this option  to the solidity compiler's option `--via-ir`.

**When to use it?**  
When we want to enable the IR-based code generator

**Example**
`certoraRun Bank.sol --verify Bank:Bank.spec --solc_via_ir`

### `--solc_evm_version`

**What does it do?**  
Passes the value of this option  to the solidity compiler's option `--evm-version`.

**When to use it?**  
When we want to select the Solidity compiler EVM version

**Example**
`certoraRun Bank.sol --verify Bank:Bank.spec --solc_evm_version Istanbul`

### `--solc_allow_path`

**What does it do?**  
Passes the value of this option as is to the solidity compiler's option `--allow-paths`.
See [--allow-path specification](https://docs.soliditylang.org/en/v0.8.16/path-resolution.html#allowed-paths)

**When to use it?**  

> When we want all an additional location for loading sources

Reviewer: ```suggestion
When we want to add an additional location the Solidity compiler to load sources from
```

`certoraRun --version`

### `--typecheck_only`

**What does it do?**  
Stops after running the Solidity compiler and type checking of the spec, before submitting the verification task.

**When to use it?**  
If you want only to check your spec, or include it in an automated task (e.g., a git `pre-commit` hook).  
**Example**

`certoraRun Bank.sol --verify Bank:bank.spec --typecheck_only`

Advanced options
----------------

(--cloud)=

> ### `--cloud`

Reviewer 3: You removed this header and its reference, so all usages of that reference ned to be updated.

**What does it do?**

Runs the Prover on the cloud.  Note that for non-Certora users, `--cloud` is
the default, so this option does nothing.

**When to use it?**

If you are a Certora employee who usually runs the Prover locally, but want to
run on the cloud instead.

(--staging)=

> ### `--staging [branch]`

Reviewer 3: ditto

**Example**

`--java_args '"-Dcvt.default.parallelism=2"'` - will set the number of “tasks” that can run in parallel to 2.

(--prover_args)=

> ### `--prover_args`

Reviewer 3: this is the header you changed, and more importantly notice that in line 655 you changed the string used to reference this header from other files.

The `--prover_args` option allows you to provide fine-grained tuning options to the
Prover.  `--prover_args` receives a string containing prover flags, the prover flags will be sent as it to the prover.

> `--prover_args` cannot set prover options that are set by standalone certoraRun options (e.g. the prover option`--t` is

Reviewer: ```suggestion
`--prover_args` cannot set prover options that are set by standalone certoraRun options (e.g. the prover option `--t` is
```

> set by `--smt_timeout` therefore cannot appear in `--prover_args`). `--prover_args` value must be quoted

Reviewer: ```suggestion
set by `--smt_timeout` therefore cannot appear in `--prover_args`). `--prover_args` value must be quoted.
```

(-optimisticReturnsize)=
#### `--prover_args '-optimisticReturnsize'`

This option determines whether {ref}`havoc summaries <havoc-summary>` assume
that the called method returns the correct number of return values.

(-showInternalFunctions)=
#### `--prover_args '-showInternalFunctions'`

A single occurrence of `--prover_args` can set multiple values

> #### `--prover_args '-showInternalFunctions -optimisticReturnsize'`

Author: I do not understand what header, I just added additional example
